src/pages/Compare/index.spec.tsx

```diff
@@ -886,6 +916,287 @@ describe(__filename, () => {
     expect(dispatch).not.toHaveBeenCalled();
   });
 
+  it('dispatches fetchDiff and fetchVersionFile for the next file with diff', () => {
+    const addonId = 10;
+    const baseVersionId = 44;
+    const headVersionId = 55;
+    const file1 = 'file1.json';
+    const file2 = 'file2.json';
+    const version = createExternalVersionWithEntries(
+      [
+        { path: file1, status: 'M' },
+        { path: 'fileWithoutChange', status: '' },
+        { path: file2, status: 'M' },
+      ],
+      { id: headVersionId },
+      // eslint-disable-next-line @typescript-eslint/camelcase
+      { selected_file: file1, diff: fakeExternalDiff },
+    ) as ExternalVersionWithDiff;
+
+    const {
+      _fetchDiff,
+      _fetchVersionFile,
+      dispatchSpy,
+      fakeThunk,
+    } = loadDiffAndRender({
+      addonId,
+      baseVersionId,
+      headVersionId,
+      version,
+      buildTree: true,
+    });
+
+    expect(dispatchSpy).toHaveBeenCalledWith(fakeThunk.thunk);
+    expect(_fetchDiff).toHaveBeenCalledWith({
+      addonId,
+      baseVersionId,
+      headVersionId,
+      path: file2,
+    });
+    expect(_fetchVersionFile).toHaveBeenCalledWith({
+      addonId,
+      versionId: headVersionId,
+      path: file2,
+    });
+  });
+
+  it('does not dispatch fetchDiff or fetchVersionFile for the next file if fileTree is not built', () => {
+    const addonId = 10;
+    const baseVersionId = 44;
+    const headVersionId = 55;
+    const file1 = 'file1.json';
+    const file2 = 'file2.json';
+    const version = createExternalVersionWithEntries(
+      [
+        { path: file1, status: 'M' },
+        { path: 'fileWithoutChange', status: '' },
+        { path: file2, status: 'M' },
+      ],
+      { id: headVersionId },
+      // eslint-disable-next-line @typescript-eslint/camelcase
+      { selected_file: file1, diff: fakeExternalDiff },
+    ) as ExternalVersionWithDiff;
+
+    const {
+      _fetchDiff,
+      _fetchVersionFile,
+      dispatchSpy,
+      fakeThunk,
+    } = loadDiffAndRender({
+      addonId,
+      baseVersionId,
+      headVersionId,
+      version,
+      buildTree: false,
+    });
+
+    expect(dispatchSpy).not.toHaveBeenCalledWith(fakeThunk.thunk);
+    expect(_fetchDiff).not.toHaveBeenCalledWith({
+      addonId,
+      baseVersionId,
+      headVersionId,
+      path: file2,
+    });
+    expect(_fetchVersionFile).not.toHaveBeenCalledWith({
+      addonId,
+      versionId: headVersionId,
+      path: file2,
+    });
+  });
+
+  it.each([['versionFile', 'loadVersionFile'], ['diff', 'loadDiff']])(
+    'does not dispatch fetchDiff or fetchVersionFile for the next file when the current %s is not loaded',
```

Comment:
How about 'it does not dispatch fetchDiff or fetchVersionFile for the next file before %s is dispatched' ? With that, you wouldn't have to pass `data` through `it.each()` -- it doesn't appear that `data` is used for anything besides the message, right?